crowbar_framework/app/controllers/api/restart_management_controller.rb

```diff
@@ -92,8 +91,17 @@ def restarts_post
       next unless node.key? :crowbar_wall
       next unless node[:crowbar_wall].key? :requires_restart
       next unless node[:crowbar_wall][:requires_restart].key? cookbook
-      next unless node[:crowbar_wall][:requires_restart][cookbook].key? service
-      node[:crowbar_wall][:requires_restart][cookbook].delete(service)
+
+      # clear all services or, clear all services of a cookbook
+      if service.nil? || cookbook == service
+        node[:crowbar_wall][:requires_restart][cookbook].each_key do |the_service|
+          node[:crowbar_wall][:requires_restart][cookbook].delete(the_service)
+        end
```

Comment:
Can't we just do something like  `node[:crowbar_wall][:requires_restart].delete(cookbook)`?

Or possibly the following which rewrites the logic in the method (not necessarily better since it's harder to read, actually):

```ruby
if service.nil?
   node[:crowbar_wall].delete(:requires_restart)
else
  managed_cookbooks.each do |cookbook|
    [...]
    if cookbook == service
      node[:crowbar_wall][:requires_restart].delete(cookbook)
    elsif node[:crowbar_wall][:requires_restart][cookbook].key? service
      node[:crowbar_wall][:requires_restart][cookbook].delete(service)
    end
  [...]
`

Reply:
I think it's a good optimization, let's change it

Reply:
This part could be outside of the loop
 
next unless node.key? :crowbar_wall
next unless node[:crowbar_wall].key? :requires_restart